pkg/controller/egress/controller.go

```diff
@@ -449,3 +451,41 @@ func (c *EgressController) enqueueEgresses(poolName string) {
 		c.queue.Add(egress.Name)
 	}
 }
+
+// updateEgressStatusCondition updates a Condition in an Egress's Status.
+func (c *EgressController) updateEgressStatusCondition(obj interface{}, statusType egressv1alpha2.EgressConditionType, statusBool bool) {
```

Comment:
should avoid using `interface{}` when possible. The reason why some lib functions use it because they are generic and can be used for many types.